diff -urNp -x '*.orig' Wnn6-SDK-1.0.org/src/contrib/im/Xsi/Wnn/jlib/Imakefile Wnn6-SDK-1.0/src/contrib/im/Xsi/Wnn/jlib/Imakefile
--- Wnn6-SDK-1.0.org/src/contrib/im/Xsi/Wnn/jlib/Imakefile	2000-09-01 11:58:55.000000000 +0200
+++ Wnn6-SDK-1.0/src/contrib/im/Xsi/Wnn/jlib/Imakefile	2023-02-02 20:06:25.656502010 +0100
@@ -4,7 +4,7 @@ XCOMM
 
 #ifdef NEED_DEFINE_LIB
 #define DoNormalLib YES
-#define DoSharedLib NO
+#define DoSharedLib YES
 #define DoDebugLib NO
 #define DoProfileLib NO
 #endif
@@ -63,7 +63,7 @@ OBJS= $(OBJ1) $(LOCAL_ETCOBJS) $(LOCAL_R
 
 
 LibraryObjectRule()
-AllTarget($(OBJS) libwnn6_fromsrc.a)
+AllTarget($(OBJS) libwnn6.a libwnn6.so.1.0.0)
 
 
 js.o : js.c \
@@ -74,13 +74,15 @@ js.o : js.c \
 
 SingleProgramTarget(kankana,kankana.o,$(WNNJLIB),)
 
-NormalLibraryTarget(wnn6_fromsrc,$(OBJS))
-LintLibraryTarget(wnn6_fromsrc,$(SRCS))
+NormalLibraryTarget(wnn6,$(OBJS))
+SharedLibraryTarget(wnn6,1.0.0,$(OBJS),.,.)
+LintLibraryTarget(wnn6,$(SRCS))
 
 instlib:: install
 
 
-InstallLibrary(wnn6_fromsrc,$(WNNLIBDIR))
+InstallLibrary(wnn6,$(WNNLIBDIR))
+InstallSharedLibrary(wnn6,1.0.0,$(WNNLIBDIR))
 
 #if ProfileLibJlib
 ProfiledLibraryTarget(wnn,$(OBJS))
