diff -ruN src.org/contrib/im/Xsi/Wnn/include/config.h src/contrib/im/Xsi/Wnn/include/config.h
--- src.org/contrib/im/Xsi/Wnn/include/config.h	Wed Oct 28 13:36:24 1998
+++ src/contrib/im/Xsi/Wnn/include/config.h	Wed Jan 27 11:36:40 1999
@@ -140,7 +140,7 @@
 #define _CONFIG_H_
 
 #ifndef	LIBDIR
-#define	LIBDIR		"/usr/local/lib/wnn6"
+#define	LIBDIR		"/var/lib/wnn6"
 #endif	/* LIBDIR */
 
 #ifdef TAIWANESE
diff -ruN src.org/contrib/im/Xsi/config/X11.tmpl src/contrib/im/Xsi/config/X11.tmpl
--- src.org/contrib/im/Xsi/config/X11.tmpl	Wed Oct 28 13:36:37 1998
+++ src/contrib/im/Xsi/config/X11.tmpl	Wed Jan 27 11:36:14 1999
@@ -7,9 +7,9 @@
 #include "/usr/openwin/lib/config/Project.tmpl"
 #else	/* Sun & SVR4 */
 #if defined(LinuxArchitecture) || defined(FreeBSDArchitecture) || defined(NetBSDArchitecture) || defined(OpenBSDArchitecture) || defined(BSD386Architecture)
-#include "/usr/X11R6/lib/X11/config/X11.tmpl"
+#include "/usr/lib64/X11/config/X11.tmpl"
 #else	/* Linux || *BSD */
-#include "/usr/lib/X11/config/Project.tmpl"
+#include "/usr/lib64/X11/config/Project.tmpl"
 #endif	/* Linux || *BSD */
 #endif	/* Sun & SVR4 */
 
@@ -60,9 +60,9 @@
 #define BinDir Concat(ProjectRoot,/bin)
 #else
 #if ((defined LINUX) || (defined FREEBSD) || (defined BSDOS) || (defined NETBSD))
-#define BinDir /usr/X11R6/bin
+#define BinDir /usr/bin
 #else
-#define BinDir /usr/bin/X11
+#define BinDir /usr/bin
 #endif /* LINUX || FREEBSD || BSDOS || NETBSD */
 #endif /* ProjectRoot */
 #endif /* BinDir */
@@ -73,7 +73,7 @@
 #define BuildIncTop ..		/* relative to $(TOP)/X11 */
 #endif
 #ifndef LibDir
-#define LibDir $(USRLIBDIR)/X11
+#define LibDir $(USRLIBDIR)
 #endif
 #ifndef ConfigDir
 #define ConfigDir $(LIBDIR)/config
@@ -502,32 +502,36 @@
 #define WnnOwner	wnn
 #endif
 
+#ifndef InstallHeader
+#define InstallHeader YES
+#endif
+
 #ifndef ImTop
 #define ImTop	$(CONTRIBSRC)/im/Xsi
 #endif
 
 #ifndef WnnWnnDir
-#define WnnWnnDir	/usr/local/lib/wnn6
+#define WnnWnnDir	/var/lib/wnn6
 #endif
 
 
 #ifndef WnnBinDir
-#define WnnBinDir	/usr/local/bin
+#define WnnBinDir	/usr/bin
 #endif
 
 #ifndef WnnIncDir
 #define WnnIncDir	$(INCROOT)
 #endif
 
 #ifndef WnnLibDir
 #define WnnLibDir	$(USRLIBDIR)
 #endif
 
        WNNOWNER = WnnOwner
           IMTOP = ImTop
       WNNWNNDIR = WnnWnnDir
       WNNBINDIR = WnnBinDir
-      WNNINCDIR = WnnIncRoot
+      WNNINCDIR = WnnIncDir
       WNNLIBDIR = WnnLibDir
 
     WNNTERM_LIB = termLibrary
@@ -539,7 +543,7 @@
      JWNNWNNDIR = $(WNNWNNDIR)/ja_JP
      JWNNBINDIR = $(WNNBINDIR)/Wnn6
 
-     JWNNINCDIR = $(WNNINCDIR)/wnn
+     JWNNINCDIR = $(WNNINCDIR)/wnn6
      JWNNDICDIR = $(JWNNWNNDIR)/dic
   JWNNPUBDICDIR = $(JWNNDICDIR)/pubdic
 JWNNWNNCONSDICDIR = $(JWNNDICDIR)/wnncons
@@ -563,7 +567,7 @@
    WNNCNVFILEDIR = $(WNNJDSRC)/cvtfileDir
     WNNINCLUDES = -I$(WNNINCLUDESRC) -I$(WNNROMKANMSRC)
      WNNDEFINES = -DLIBDIR=\"$(WNNWNNDIR)\"
-        WNNJLIB = $(WNNJLIBSRC)/libwnn6_fromsrc.a
+        WNNJLIB = $(WNNJLIBSRC)/libwnn6.a
      WNNCONVLIB = $(WNNCONVSRC)/libconvkey.a
 
         CWNNTOP = $(IMTOP)/cWnn
diff -ur src/contrib/im/Xsi/Makefile.ini src.new/contrib/im/Xsi/Makefile.ini
--- src/contrib/im/Xsi/Makefile.ini	2011-07-14 11:38:14.915761854 +0200
+++ src.new/contrib/im/Xsi/Makefile.ini	2011-07-14 11:31:20.392761947 +0200
@@ -15,7 +15,7 @@
       CURRENT_DIR = ./../contrib/im/Xsi
 #       CONFIGSRC = /usr/lib/X11/config
 #       CONFIGSRC = /usr/openwin/lib/X11/config
-        CONFIGSRC = /usr/X11R6/lib/X11/config
+        CONFIGSRC = /usr/lib64/X11/config
          IMAKESRC = $(CONFIGSRC)
          IRULESRC = $(CONFIGSRC)
           WSFLAGS = 
--- src/contrib/im/Xsi/config/Project.tmpl~	2000-09-01 11:58:55.000000000 +0200
+++ src/contrib/im/Xsi/config/Project.tmpl	2011-07-14 12:14:57.879151945 +0200
@@ -7,9 +7,9 @@
 #include "/usr/openwin/lib/config/Project.tmpl"
 #else	/* Sun & SVR4 */
 #if defined(LinuxArchitecture) || defined(FreeBSDArchitecture) || defined(NetBSDArchitecture) || defined(OpenBSDArchitecture) || defined(BSD386Architecture)
-#include "/usr/X11R6/lib/X11/config/Project.tmpl"
+#include "/usr/lib64/X11/config/Project.tmpl"
 #else	/* Linux || *BSD */
-#include "/usr/lib/X11/config/Project.tmpl"
+#include "/usr/lib64/X11/config/Project.tmpl"
 #endif	/* Linux || *BSD */
 #endif	/* Sun & SVR4 */
 
