--- Xsi/Wnn/jlib/Imakefile.orig	Wed Oct 28 13:36:30 1998
+++ Xsi/Wnn/jlib/Imakefile	Tue Nov 10 00:52:15 1998
@@ -4,7 +4,7 @@
 
 #ifdef NEED_DEFINE_LIB
 #define DoNormalLib YES
-#define DoSharedLib NO
+#define DoSharedLib YES
 #define DoDebugLib NO
 #define DoProfileLib NO
 #endif
@@ -61,7 +61,7 @@
 OBJS= $(OBJ1) $(LOCAL_ETCOBJS) $(LOCAL_RKOBJS)
 
 LibraryObjectRule()
-AllTarget($(OBJS) libwnn6_fromsrc.a)
+AllTarget($(OBJS) libwnn6.a libwnn6.so.1.0.0)
 
 
 js.o : js.c \
@@ -72,13 +72,15 @@
 
 SingleProgramTarget(kankana,kankana.o,$(WNNJLIB),)
 
-NormalLibraryTarget(wnn6_fromsrc,$(OBJS))
-LintLibraryTarget(wnn6_fromsrc,$(SRCS))
+NormalLibraryTarget(wnn6,$(OBJS))
+SharedLibraryTarget(wnn6,1.0.0,$(OBJS),.,.)
+LintLibraryTarget(wnn6,$(SRCS))
 
 instlib:: install


-InstallLibrary(wnn6_fromsrc,$(WNNLIBDIR))
+InstallLibrary(wnn6,$(WNNLIBDIR))
+InstallSharedLibrary(wnn6,1.0.0,$(WNNLIBDIR))
 
 #if ProfileLibJlib
 ProfiledLibraryTarget(wnn,$(OBJS))
